lib/pmdtester/builders/diff_builder.rb

```diff
@@ -17,6 +17,27 @@ def build(base_report, patch_report)
       report_diff
     end
 
+    # Serving single mode
+    def build_single(patch_report)
```

Comment:
Hm.... build and build_single are almost identical. Have you thought about using a default parameter? like

```
def build(base_report = nil, patch_report)
```

Hm... looks like, we would need a condition then inside, depending on whether base_report is given or not...